plugins/csi/plugin.go

```diff
@@ -500,6 +468,28 @@ type VolumeContentSource struct {
 	CloneID    string
 }
 
+func (vcr *VolumeContentSource) ToCSIRepresentation() *csipbv1.VolumeContentSource {
+	if vcr == nil {
+		return nil
+	}
+	result := &csipbv1.VolumeContentSource{}
+	if vcr.CloneID != "" {
+		result.Type = &csipbv1.VolumeContentSource_Volume{
+			Volume: &csipbv1.VolumeContentSource_VolumeSource{
+				VolumeId: vcr.CloneID,
+			},
+		}
+	}
+	if vcr.SnapshotID != "" {
```

Comment:
I'd suggest using `else` here to emphasize that these are mutual exclusive conditions. Also, I'd suggest adding a comment to the struct indicating that exactly one field is populated.

```suggestion
	} else if vcr.SnapshotID != "" {
```

Also, want to double check our future proofing story: Given that Nomad originates `VolumeContentSource`, if CSI adds a new type, we'll simply not support it until we update the code. Is that true?

Reply:
Correct. Which is safe so long as the spec doesn't change such that one of the fields _must_ be set, but that would break every existing plugin out there so I don't suspect we'd see that kind of change.

Reply:
Addressed in c9a6ab7